src/flow_graph.py

```diff
@@ -312,10 +311,10 @@ def process_with_delay_slots(item: Union[Instruction, Label]) -> None:
             block_builder.set_label(item)
             return
 
-        if not arch.is_delay_slot_instruction(item):
+        if not item.has_delay_slot:
             block_builder.add_instruction(item)
             # Split blocks at jumps, at the next instruction.
-            if arch.is_jump_instruction(item):
+            if item.is_jump():
                 block_builder.new_block()
```

Comment:
pre-existing: I was a bit confused at this check: turns out it's always false and I guess added for compatibility with hypothetical future arches with delay slots and delay-slot-free branches. Turn it into an assert, I guess. And maybe rename the function process_mips since it has some mips-specific stuff in it.

Btw, the
```
# (Best-effort check for whether the instruction can be
# executed twice in a row.)
```
heuristic below is perhaps something we can improve with inputs/outputs for Instruction's.

Reply:
Oh, that makes a lot of sense. I was also a bit confused about this when implementing PPC, but kind of worked around it. 
I'll fix that, rename the fn, and add a `TODO` about the heuristic.

I'm also going to remove `ArchAsm.uses_delay_slots` and use `arch.arch == MIPS` instead.